### PR TITLE
Adding all ASI RP APIs

### PR DESCRIPTION
**Pull request by nirgfani**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 1]
```diff
@@ -256,18 +256,61 @@
                 }
             }
         },
-        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/dataConnectors": {
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/alertRules/{ruleId}/actions": {
             "get": {
-                "x-ms-examples": {
-                    "Get all data connectors.": {
-                        "$ref": "./examples/dataConnectors/GetDataConnectors.json"
+                "x-ms-examples": {},
```
Comment (jhendrixMSFT) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: There are a lot of empty `x-ms-examples`, when will the missing examples be added?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 2]
```diff
@@ -464,559 +689,2050 @@
...
+                    "description": "The severity of the case",
+                    "enum": [
+                        "Critical",
+                        "High",
+                        "Medium",
+                        "Low",
+                        "Informational"
+                    ],
+                    "x-ms-enum": {
+                        "name": "severity",
```
Comment (jhendrixMSFT) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: You have duplicate `x-ms-enum` types with different sets of values which is not allowed, see other definition [here](https://github.com/Azure/azure-rest-api-specs/blob/4cd8dd6c83ae0e8a06653691b3b6993568e81164/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json#L1599).  Either change the name or ensure both enums contain the same set of values.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Still waiting on ARM review however enum must be fixed.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json (hunk 1 above)
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: The example was in the folder but the references were missing (fix it)

**Comment by jhendrixMSFT** [comment]
There's another `x-ms-enum` name collision `status`.  See lines 1917 and 1948.

**Comment by majastrz** [comment]
>                 },

"tags" is a top-level resource property and shouldn't be duplicated inside the properties envelope. Are you creating your own concept of tags separate from ARM or are these the usual ARM tags?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:1856 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                 },

This is going to cause confusion with the ARM concept of tags. ARM tags are a top-level resource property and are an object instead of an array. Can we rename it to something else on all the resource types?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:1856 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>         "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/actions": {

Where's the corresponding GET /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/actions/{actionId}?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:468 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>         "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/cases": {

This applies to all the extension resource types: What error message will the user get if they attempt to create the extension resource type at a scope that you don't support (something that's not an LA workspace)?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:516 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                 },

(This comment applies to all the resource types in the PR that have this property.)

---
In reply to: [462465281](https://github.com/Azure/azure-rest-api-specs/pull/5173#issuecomment-462465281) [](ancestors = 462465281)

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:1856 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                                 "description": "Case is in closed status"

This is non-blocking and can be addressed later, but these enum comments aren't really meaningful. I can tell it's in closed status when the status property is set to "Closed". I'd elaborate about implications of each value if possible.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:1934 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                     "description": "Bookmark data",

nit - properties

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2008 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                         }

This should probably reference the UserInfo type.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2052 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                 "displayName"

It seems to be more properties (query?) should be required - otherwise, what is it really bookmarking?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2072 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                                 "value": "Account"

You should add descriptions for auto-generated docs.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2432 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>             "type": "object",

Should this be string?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2416 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>             "x-ms-discriminator-value": "Account",

Have you verified that the SDK generated by the Swagger is capable of deserializing Entities of all kinds?

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2446 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                     "x-ms-enum": {

This is lower pri, but you should add descriptions.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2573 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>             "description": "Represents settings for UEBA enablement.",

Please expand the acronym in all the descriptions. It will show in generated REST docs.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2730 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                         "UebaSettings",

Please consider expanding the acronym on the enum value. UserAndEntityBehaviorAnalyticsSettings is probably too much, but BehaviorAnalyticsSettings would improve readability.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2717 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                     "description": "Determines whether the tenant .",

Incomplete description

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2768 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                     "description": "Determines whether UEBA is enabled from MCAS.",

Please expand the acronym.

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2755 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Comment by majastrz** [comment]
>                     "description": "Determines whether the tenant .",

Also please expand ATP in the description once you add it.

---
In reply to: [462482230](https://github.com/Azure/azure-rest-api-specs/pull/5173#issuecomment-462482230) [](ancestors = 462482230)

---
Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2768 in 36b0f2d. [](commit_id = 36b0f2d1546045298331a242d98069c0e6af3c0c, deletion_comment = False)

**Review by majastrz — CHANGES_REQUESTED** [review]
:clock1:

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 3]
```diff
@@ -464,559 +464,2343 @@
...
+                    "description": "Scheduled alert rule properties",
+                    "$ref": "#/definitions/ScheduledAlertRuleProperties"
+                }
+            }
+        },
+        "ScheduledAlertRuleProperties": {
+            "type": "object",
+            "description": "Alert rule property bag.",
+            "properties": {
+                "ruleName": {
```
Comment (ravbhatnagar) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: ruleName comes from the URL and is also a top level property. Why is this needed?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 4]
```diff
@@ -464,559 +464,2343 @@
...
+                        "name": "TriggerOperator",
+                        "modelAsString": false
+                    },
+                    "description": "The operation against the threshold that triggers alert rule."
+                },
+                "triggerThreshold": {
+                    "type": "integer",
+                    "description": "The threshold triggers this alert rule."
+                },
+                "suppressionEnabled": {
```
Comment (ravbhatnagar) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: bools are typically not recommended. two reasons - 1. they are less descriptive than string enums 2. they dont allow for future expansion of possible states if needed. Recommended way would be to have string enum. ex- suppressionState suppressionStatus with possible values as enabled or disabled. 
Same comment applies to all other bools that you have elsewhere.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 5]
```diff
@@ -464,559 +464,2343 @@
...
+                    "type": "string",
+                    "format": "date-time",
+                    "description": "The end time of the case"
+                },
+                "startTimeUtc": {
+                    "type": "string",
+                    "format": "date-time",
+                    "description": "The start time of the case"
+                },
+                "tags": {
```
Comment (ravbhatnagar) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: everywhere, tags should be a top level property. Is it modeled like this since tags on proxy resource is not supported?

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

**Comment by nirgfani** [comment]
> > ```
> >     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/cases": {
> > ```
> 
> This applies to all the extension resource types: What error message will the user get if they attempt to create the extension resource type at a scope that you don't support (something that's not an LA workspace)?
> 
> Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:516 in [36b0f2d](https://github.com/Azure/azure-rest-api-specs/commit/36b0f2d1546045298331a242d98069c0e6af3c0c). [](commit_id = [36b0f2d](https://github.com/Azure/azure-rest-api-specs/commit/36b0f2d1546045298331a242d98069c0e6af3c0c), deletion_comment = False)

Without LA workspace it won't exist, in our code we are listening only on request that are being made on Microsoft.OperationalInsights RP-/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalIinsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/cases

**Comment by nirgfani** [comment]
@majastrz, @ravbhatnagar - I see that you both have questions regarding "actions", let's discuss it today on the meeting

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json (hunk 3 above)
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: The ruleName is a display name that the customer is giving to the alert, it has spaces and have no restrictions.
The top level "name" property is the one that any resource has in the system, we are planing to generate a guid from our UX since it easier to handle as unique identifier of the resource.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json (hunk 4 above)
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: We know it, and we made enums where there was question whether in the future there will be more options, in this case as well as other cases where we kept it boolean it will make no sense.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json (hunk 5 above)
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: It will have a bit different behavior than regular tags, and in order to avoid the confusion we will change it to labels

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 6]
```diff
@@ -256,18 +256,18 @@
                 }
             }
         },
```
[the remark this answers is not in the document]
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: This is a good feedback, but patch won't be supported at first phase.
We are starting with the basic API, and there is full UX experience that is using it. We might extend it in the future if needed.

Regarding the PUT- it will require to send the entire resource each time, meaning that in your example at the third PUT it will unassign.

**Comment by nirgfani** [comment]
> > ```
> >         "x-ms-discriminator-value": "Account",
> > ```
> 
> Have you verified that the SDK generated by the Swagger is capable of deserializing Entities of all kinds?
> 
> Refers to: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:2446 in [36b0f2d](https://github.com/Azure/azure-rest-api-specs/commit/36b0f2d1546045298331a242d98069c0e6af3c0c). [](commit_id = [36b0f2d](https://github.com/Azure/azure-rest-api-specs/commit/36b0f2d1546045298331a242d98069c0e6af3c0c), deletion_comment = False)

What's the different between it and other cases? What's your concern?
How do I run it? I ran the AutoReset and it pass.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 7]
```diff
@@ -464,559 +464,2343 @@
...
-    },
-    "definitions": {
-        "operation": {
-            "description": "Operation provided by provider",
-            "properties": {
-                "name": {
-                    "type": "string",
-                    "description": "Name of the operation"
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/actions": {
```
[the remark this answers is not in the document]
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: As we agreed on the call I will remove the top level resource, and we will keep for now only the nested resource. We will see by customers feedback if the top level resource is needed.
Once we will need to extend this capability to cases or other we will reconsider the options.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 8]
```diff
@@ -464,559 +464,2343 @@
...
+        },
+        "ActionProperties": {
+            "type": "object",
+            "description": "Action property bag.",
+            "properties": {
+                "triggerUri": {
+                    "type": "string",
+                    "description": "The uri for the action to trigger."
+                },
+                "ruleId": {
```
[the remark this answers is not in the document]
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: As agreed since the top level resource will be removed for now, there will be no need in ruleId, so only the triggerUri will be included

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 9]
```diff
@@ -304,30 +307,18 @@
                 }
             }
         },
-        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/dataConnectors/{dataConnectorId}": {
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/alertRules/{ruleId}/actions/{actionId}": {
```
[the remark this answers is not in the document]
Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: We will remove the pattern from all parameters

**Comment by ravbhatnagar** [comment]
One design feedback needs to be revisited - currently, actions is a child type under alertRules. This means that to associate a webhook, or endpoint with multiple alertRules, multiple child resources of type actions must be created. This will result in management overhead for these resources. In future, service team will support actions under cases resource type as well. And the plan is to add this as a nested resource type under cases as well. 
The recommendations were are follows -
1. Model actions as a top level resource type. and then it could be re-used/associated with any number of alertRules or cases
2. Model it as a property on the alertRules and cases. This may not be optimal when it needs to be an array and updating will not be straight forward.
Service team will consider this feedback after the preview release. potentially get feedback from customers and understand more how they use these actions.
SIgning off from ARM side with this note.

File: package.json [hunk 10]
```diff
@@ -30,7 +30,8 @@
     "ts-node": "^8.0.1"
   },
   "dependencies": {
-    "@octokit/rest": "^15.2.6"
+    "@octokit/rest": "^15.2.6",
+    "swagger-typescript-codegen": "^1.9.2"
```
Comment (jhendrixMSFT) [inline package.json]: Why is this change here?

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
The JSON is malformed causing parsing failure, see the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/492552656).

**Comment by jhendrixMSFT** [comment]
@nirgfani also is the service deployed and live?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 11]
```diff
@@ -365,15 +359,15 @@
...
                     }
                 },
                 "tags": [
-                    "Data Connectors"
+                    "Actions"
                 ],
-                "description": "Creates or updates the data connector.",
-                "operationId": "DataConnectors_Create",
+                "description": "Creates or updates the action of alert rule.",
+                "operationId": "AlertRules_CreateAction",
```
Comment (jhendrixMSFT) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Should be AlertRules_CreateOrUpdateAction.  Same rule applies to all operationIds with a _Create suffix.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 12]
```diff
@@ -464,559 +464,2337 @@
...
+                    "type": "string",
+                    "description": "The status of the case",
+                    "enum": [
+                        "Draft",
+                        "Open",
+                        "InProgress",
+                        "Closed"
+                    ],
+                    "x-ms-enum": {
+                        "name": "Status",
```
Comment (jhendrixMSFT) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: How about a more descriptive name, perhaps `CaseStatus`?

**Review by jhendrixMSFT — APPROVED** [review]

**Review by ravbhatnagar — APPROVED** [review]

**Review by majastrz — APPROVED** [review]
